src/WriteLockStrategy/MariaDbMetadataLockStrategy.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * This file is part of prooph/pdo-event-store.
+ * (c) 2016-2019 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2019 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Pdo\WriteLockStrategy;
+
+use Prooph\EventStore\Pdo\WriteLockStrategy;
+
+final class MariaDbMetadataLockStrategy implements WriteLockStrategy
+{
+    /**
+     * @var \PDO
+     */
+    private $connection;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+
+    public function __construct(\PDO $connection, int $timeout = 0xffffff)
+    {
+        if ($timeout < 0) {
```

Comment:
```suggestion
        if ($timeout <= 0) {
```

Reply:
0 should work as a timeout I think

Reply:
@fritz-gerneth that statement doesn't go with the exception, you stated in the exception `$timeout must be larger then 0.`

Reply:
Fixed the exception message with the latest commit now